flink-ml-iteration/src/main/java/org/apache/flink/iteration/operator/OperatorUtils.java

```diff
@@ -136,31 +135,20 @@ public static StreamConfig createWrappedOperatorConfig(StreamConfig config, Clas
         wrappedConfig.setTypeSerializerOut(
                 ((IterationRecordSerializer<?>) typeSerializerOut).getInnerSerializer());
 
-        Stream.concat(
-                        config.getChainedOutputs(cl).stream(),
-                        config.getNonChainedOutputs(cl).stream())
+        config.getChainedOutputs(cl)
                 .forEach(
-                        edge -> {
-                            OutputTag<?> outputTag = edge.getOutputTag();
-                            if (outputTag != null) {
-                                TypeSerializer<?> typeSerializerSideOut =
-                                        config.getTypeSerializerSideOut(outputTag, cl);
-                                checkState(
-                                        typeSerializerSideOut instanceof IterationRecordSerializer,
-                                        "The serializer of side output with tag[%s] should be IterationRecordSerializer but it is %s.",
-                                        outputTag,
-                                        typeSerializerSideOut);
-                                wrappedConfig.setTypeSerializerSideOut(
-                                        new OutputTag<>(
-                                                outputTag.getId(),
-                                                ((IterationRecordTypeInfo<?>)
-                                                                outputTag.getTypeInfo())
-                                                        .getInnerTypeInfo()),
-                                        ((IterationRecordSerializer) typeSerializerSideOut)
-                                                .getInnerSerializer());
-                            }
+                        chainedOutput -> {
+                            OutputTag<?> outputTag = chainedOutput.getOutputTag();
+                            setTypeSerializerSideOut(outputTag, config, wrappedConfig, cl);
+                        });
+        config.getOperatorNonChainedOutputs(cl)
```

Comment:
Would it be simpler to keep the previous style and do something like this:

```
Stream.concat(
        config.getChainedOutputs(cl).stream(),
        config.getOperatorNonChainedOutputs(cl).stream())
        .forEach(
                output -> {
                    OutputTag<?> outputTag = output.getOutputTag();
                    setTypeSerializerSideOut(outputTag, config, wrappedConfig, cl);
                }
        );
```

Reply:
The reason why I separate these two parts is `config.getChainedOutputs(cl)` returns `List<StreamEdge>` while `config.getOperatorNonChainedOutputs(cl)` returns `List<NonChainedOutput>`, they are of different types and are not suitable to iterate in the same loop.